### DISCUSSION TITLE
.Net live session follow-up

### DISCUSSION THREAD

**Discussion by melursus23** [ScottArbeit/Grace#10] [General]
Hello,

I want to congratulate you for your ambitious project! 
As a C# developer who's trying to learn F# on and off for the last 2 years, I'm always curious to explore "real-world" project as I've always struggle to find some. I'm secretly dreaming that the F# community has a killer app that will make obvious that F# is a great language, no pressure!
It will be interesting if one day you do a more technical presentation of your solution!

I'm also waiting for the video demo promised where we can see the whole flow from the client to the server :-)

https://www.youtube.com/watch?v=4UmDOry4SnA&themeRefresh=1

**Comment by ScottArbeit** [comment]
Thanks, @melursus23! 

After Monday's technical issues on-stream, (even though I don't think it was my computer causing it) I finally bought a new computer with a good GPU and I'll be making videos very soon, both for the repo and for YouTube. You can expect some very technical deep-dives.

I have hopes that Grace can be that killer app for F#. There's definitely a growing awareness in the industry that functional constructs mean better code, and F# makes most of them available.

I'll also be presenting at [NDC Oslo 2023](https://ndcoslo.com/agenda/what-comes-after-git-or-building-a-cloud-native-version-control-system-in-f-04vz/02j8fse9bsm) at the end of May.

**Reply by houstonhaynes** [reply]
FYI this is one of my favorite clips from Mads Torgerson's presentation on the "functional journey of C#" - he basically digresses into how OOP is "screwed" [*his* word] in new cloud-based scenarios. Interesting talk all the way through... https://www.youtube.com/watch?v=CLKZ7ZgVido&t=2823s

**Reply by ScottArbeit** [reply]
Obviously, I agree. To be clear, I still use objects (i.e. classes) in Grace, especially when I'm interacting with NuGet packages (where the usual inherit-from-this-class or implement-this-interface still applies) but I treat them functionally and as if they're immutable.

For years I heard "learning F# will make you a better C# programmer" and it's completely true.

AND with F# you get access to constructs that make more beautiful code.

**Reply by houstonhaynes** [reply]
💯 On C# -> F# ... I received some coaching from Andrii Chebukin who built his own actor model in F# for a rideshare system he built in Eastern Europe. I was having some trouble with getting Redis build on-the-fly indices (in Redis.OM - which insisted on naming a single index in the record decorator) and very casually said "let's just look at the C# and build an equivalent in F# *without* the problem" 😨  😁  That was my gateway drug... I'm working on extending an Elmish.Avalonia "example" app and found myself passing around objects without really thinking too much about it - and to be fair I'm standing on some pretty broad shoulders there - so it's not like I crafted it from a cold start. Great stuff!
